cpp/src/arrow/csv/csv-to-arrow.cc

```diff
@@ -0,0 +1,162 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
For clarity I'd either name this file something like `cli-csv-to-arrow.cc` or put it in a `arrow/cli` directory.

Reply:
We have the tools directory https://github.com/apache/arrow/tree/master/cpp/tools so this would be a good place